### ISSUE TITLE
Set-constant scriptlet issue on youtube

### ISSUE THREAD

**Issue by llacb47** [uBlockOrigin/uBlock-issues#1162]
### Prerequisites

- [x] I verified that this is not a filter issue
    - Filter issues MUST be reported at [filter issue tracker](https://github.com/uBlockOrigin/uAssets/issues)
    
- [x] This is not a support issue or a question
    - Support issues and questions are handled at [/r/uBlockOrigin](https://old.reddit.com/r/uBlockOrigin/)
    
- [x] I performed a cursory search of the issue tracker to avoid opening a duplicate issue
    - Your issue may already be reported.
- I tried to reproduce the issue when...
    - [x] uBlock Origin is the only extension
    - [x] uBlock Origin with default lists/settings
    - [x] using a new, unmodified browser profile
- [x] I am running the latest version of uBlock Origin
- [x] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### Description

`set-constant` scriptlet fails on youtube.

### A specific URL where the issue occurs

https://www.youtube.com/watch?v=SrPid1A-eKw

### Steps to Reproduce

1. Add `youtube.com##+js(set, ytInitialPlayerResponse.adPlacements, undefined)` to custom filters
2. Visit the video

### Expected behavior:

`ytInitialPlayerResponse.adPlacements` is undefined

### Actual behavior:

`ytInitialPlayerResponse.adPlacements` is unchanged

### Your environment

* uBlock Origin version: Tested on 1.27.10 and 1.28.2
* Browser Name and version: Chrome 84.0.4147.89 and Firefox 78.0.2 
* Operating System and version: Windows 10

### Notes

This simple tampermonkey script works: (made to verify that youtube was not overriding changes to the variable)
```
// ==UserScript==
// @name        YouTube debugging
// @namespace   Test
// @description
// @include     https://www.youtube.com/*
// @run-at      document-start
// @grant       none
// @version     1
// ==/UserScript==
(function() {
    'use strict';
    function waitForElement() {
        if (typeof ytInitialPlayerResponse !== "undefined" && !(ytInitialPlayerResponse == null)) {
            if (typeof ytInitialPlayerResponse.adPlacements !== "undefined") {
                window.ytInitialPlayerResponse.adPlacements = undefined;

            }
        } else {
            setTimeout(waitForElement, 1);
        }
    }
    waitForElement();
})();
```

**Comment by uBlock-user** [comment]
Can't reproduce on Firefox Stable/Nightly.

**Comment by gorhill** [comment]
> `ytInitialPlayerResponse.adPlacements` is unchanged

How did you come to the conclusion it is unchanged?

**Comment by llacb47** [comment]
I just tested and can still reproduce in FF Nightly 80.0a1 with uBo 1.28.2 and two different browser profiles in Chrome 84.0.4147.89.

> > `ytInitialPlayerResponse.adPlacements` is unchanged
> 
> How did you come to the conclusion it is unchanged?

I evaluated it with devtools.

**Comment by gorhill** [comment]
> I evaluated it with devtools.

That won't work, I don't think dev tools respects trapped variable. I stepped into the code and `ytInitialPlayerResponse.adPlacements` returns `undefined` as expected.

**Comment by gorhill** [comment]
Type `ytInitialPlayerResponse.adPlacements` in the console, and be sure the console selector is set to `top`.

**Comment by llacb47** [comment]
Ok, I think I discovered why nobody can reproduce. 🤦‍♂️ I was testing with two entries added:
```
youtube.com##+js(set, ytInitialPlayerResponse.playerAds, undefined)
youtube.com##+js(set, ytInitialPlayerResponse.adPlacements, undefined)
```
I think herein lies the bug.

If I comment out one of the entries but leave the other uncommented, the uncommented one works as expected. Checked in dev console in FF and Chrome.
(by itself) `youtube.com##+js(set, ytInitialPlayerResponse.playerAds, undefined)` **Works**

(by itself) `youtube.com##+js(set, ytInitialPlayerResponse.adPlacements, undefined)` **Works**


However, if I uncomment both of the entries, one of the properties targeted does not get set to undefined. It is usually the property targeted in the second line I think. My filters were in this order

```
youtube.com##+js(set, ytInitialPlayerResponse.adPlacements, undefined)
youtube.com##+js(set, ytInitialPlayerResponse.playerAds, undefined)
```

But if I switch the order 

```
youtube.com##+js(set, ytInitialPlayerResponse.adPlacements, undefined)
youtube.com##+js(set, ytInitialPlayerResponse.playerAds, undefined)
```

now `ytInitialPlayerResponse.playerAds` fails to set and `ytInitialPlayerResponse.adPlacements` is set to undefined.
